ops/charm.py

```diff
@@ -39,62 +56,160 @@ def restore(self, snapshot):
         self.params = self.framework.model._backend.action_get()
 
     def set_results(self, results):
+        """Report the result of the action.
+
+        :param results: The result of the action as a Dict
+        :type results: collections.abc.Mapping
+        :rtype: None
+        """
         self.framework.model._backend.action_set(results)
 
     def log(self, message):
+        """Send a message that a user will see while the action is running.
+
+        :param message: The message for the user.
+        :type message: str
+        :rtype: None
+        """
         self.framework.model._backend.action_log(message)
 
     def fail(self, message=''):
+        """Report that this action has failed.
+
+        :param message: Optional message to record why it has failed.
+        :type message: str
+        :rtype: None
+        """
         self.framework.model._backend.action_fail(message)
 
 
 class InstallEvent(HookEvent):
+    """Represents the `install` hook from Juju,
+    triggered by `CharmBase.on.install`"""
     pass
```

Comment:
Suggested:

```
    """Represents the `install` hook from Juju.

    It's triggered by `CharmBase.on.install`.
    """
```

Reply:
Thanks for the feedback.
I've pushed up a change that seems to make pydocstyle happy. It still complains about Imperative tense, and lack of module level doc strings, but it doesn't seem to complain about the structure of the docstrings anymore.